keys_storage.go

```diff
@@ -26,20 +27,21 @@ type KeysStorage interface {
 	Count(k Key) (uint, error)
 
 	// All returns all the keys
-	All() (map[Key]map[uint]Key, error)
+	All() (map[string]map[uint]Key, error)
 }
 
 // KeysStorageInMemory is an in-memory message keys storage.
 type KeysStorageInMemory struct {
-	keys map[Key]map[uint]InMemoryKey
+	keys map[string]map[uint]InMemoryKey
 }
 
 // Get returns a message key by the given key and message number.
 func (s *KeysStorageInMemory) Get(pubKey Key, msgNum uint) (Key, bool, error) {
+	index := fmt.Sprintf("%x", pubKey)
```

Comment:
The same can be done with more explicit:
```suggestion
	index := hex.EncodeToString(pubKey)
```
This should be faster and have smaller memory footprint (probably does not matter much in this case but why not).